### ISSUE TITLE
IPython-specific instructions in the docs

### ISSUE THREAD

**Issue by takluyver** [jupyter/notebook#1504]
These sections need to be made a bit more general:

- http://jupyter-notebook.readthedocs.io/en/latest/notebook.html#opening-notebooks
- http://jupyter-notebook.readthedocs.io/en/latest/examples/Notebook/Connecting%20with%20the%20Qt%20Console.html

**Comment by willingc** [comment]
@takluyver Just to clarify... Making these doc sections more general would include the use of languages kernels other than IPython such as R, Julia...

Correct?

**Comment by janschulz** [comment]
AFAIR the initial problem was that it talked about magics as if these would belong to the notebook and work in all kernels and not only the ipython one.

**Comment by willingc** [comment]
@janschulz Thanks for the context. 

We should make the following clearer in the docs: Whether magics work in a notebook depends on whether the magic is supported by the  language kernel. Some kernels, such as IRkernel, do support magicss.

**Comment by janschulz** [comment]
@willingc: Is there actually any other kernel than IPython which does support magics?

On the other hand, i think it would be possible to add a way to get the same information than the connect_info magic, but it will be a regular R function or a property. Is that desireable?

**Comment by takluyver** [comment]
Yeah, I think the important thing for the docs is just to make clear what
is "for example, to do this with the IPython kernel..."

For things like connection info that make sense regardless of the kernel,
it might be best to add a way to get it through the notebook ui, rather
than encouraging every kernel to recreate it.
